src/geodata2.erl

```diff
@@ -33,36 +43,57 @@ start_link(Name) ->
 stop() ->
     application:stop(geodata2).
 
-new(File) ->
-    case filelib:is_file(File) of
-        true ->
-            {ok, RawData} = file:read_file(File),
-            Data =
-                case is_compressed(File) of
-                    true ->
-                        zlib:gunzip(RawData);
-                    false ->
-                        RawData
-                end,
-            {ok, Meta} = geodata2_format:meta(Data),
-            %% @TODO [RTI-8302] This one could be removed after the IPv4+IPv6 MMDB is definitely used
-            set_is_ipv6_mmdb(Meta),
-            ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
-            ets:insert(?GEODATA2_STATE_TID, {data, Data}),
-            ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
-            {ok, #state{}};
-        _ ->
-            {stop, {geodata2_dbfile_not_found, File}}
-    end.
+new(MagellanFile, DomainFile) ->
+    MagellanState = case filelib:is_file(MagellanFile) of
+      true ->
+          {ok, RawData} = file:read_file(MagellanFile),
+          Data = case is_compressed(MagellanFile) of
+                   true ->
+                       zlib:gunzip(RawData);
+                   false ->
+                       RawData
+                 end,
+          {ok, Meta} = geodata2_format:meta(Data),
+          ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
+          ets:insert(?GEODATA2_STATE_TID, {data, Data}),
+          ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
+          {ok, #state{}};
+      _ ->
+          {stop, {geodata2_dbfile_not_found, MagellanFile}}
```

Comment:
As RTB does not use the domain files for now, in my opinion the first file is mandatory and the second one is optional so this new function might need to take it into account. The stop result that is returned in line 62 or 79 have special meaning on the init() function in a gen_server, stoppig the application, and therefore stopping dyno. They mark a strong condition in which there is a severe error and we cant follow on starting the adserver. In this case, if we dont have geodata we cant continue working so the application must stop. If you agree, lets make sure we only load the second file if the first one is loaded ok. But if there is an error the application stops.

Reply:
Understood, and I agree.